tests/snap_test.go

```diff
@@ -0,0 +1,203 @@
+	TestedGpioLine = ""
+)
+
+func init() {
+	TestedGpioChip = os.Getenv(SpecificGpioChip)
+	TestedGpioLine = os.Getenv(SpecifGpioLine)
+}
+
+var start = time.Now()
+
+const snapName = "matter-pi-gpio-commander"
+const chipToolSnap = "chip-tool"
+
+func TestMain(m *testing.M) {
+	teardown, err := setup() // THIS NEED TO receive "t" but this function doen't have access to it TODO: fix this latter
+	if err != nil {
+		log.Fatalf("Failed to setup tests: %s", err)
+	}
+
+	code := m.Run()
+	teardown()
+
+	os.Exit(code)
+}
+
+const sedCommand = `sed -i 's|/sys/devices/platform/axi/\*.pcie/\*.gpio/gpiochip4/dev|/sys/devices/platform/axi/*.pcie/*.gpio/gpiochip4/dev\n      - /sys/devices/platform/gpio-mockup.*/gpiochip*/dev|' squashfs-root/meta/snap.yaml`
+
+func authorizeGpioMock(path string) (string, error) {
+	_, stderr, err := utils.Exec(nil, "unsquashfs "+path)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	_, stderr, err = utils.Exec(nil, sedCommand)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	directory := filepath.Dir(path)
+	newPath := directory + "/mod_matter-pi-gpio-commander.snap"
+	_, stderr, err = utils.Exec(nil, "mksquashfs squashfs-root "+newPath)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	utils.Exec(nil, "rm -rf squashfs-root")
+	return newPath, nil
+}
+
+func setup() (teardown func(), err error) {
+	var newPath string
+
+	log.Println("[CLEAN]")
+	utils.SnapRemove(nil, snapName)
+
+	log.Println("[SETUP]")
+
+	teardown = func() {
+		log.Println("[TEARDOWN]")
+		utils.SnapDumpLogs(nil, start, snapName)
+
+		utils.Exec(nil, "rm "+newPath)
+
+		log.Println("Removing installed snap:", !utils.SkipTeardownRemoval)
+		if !utils.SkipTeardownRemoval {
+			utils.SnapRemove(nil, snapName)
+		}
+	}
+
+	// authorize gpio mock
+	if utils.LocalServiceSnap() {
+		newPath, err = authorizeGpioMock(utils.LocalServiceSnapPath)
+	}
+	if err != nil {
+		teardown()
+		return
+	}
+
+	// install matter-pi-gpio-commander
+	if utils.LocalServiceSnap() {
+		err = utils.SnapInstallFromFile(nil, newPath)
+	} else {
+		err = utils.SnapInstallFromStore(nil, snapName, utils.ServiceChannel)
+	}
+	if err != nil {
+		teardown()
+		return
+	}
+
+	// install chip-tool
+	err = utils.SnapInstallFromStore(nil, chipToolSnap, utils.ServiceChannel)
+	if err != nil {
+		teardown()
+		return
+	}
+
+	// connect interfaces:
+
+	// matter-pi-gpio-commander interfaces
+	utils.SnapConnect(nil, snapName+":avahi-control", "")
+	utils.SnapConnect(nil, snapName+":bluez", "")
+	utils.SnapConnect(nil, snapName+":network", "")
+	utils.SnapConnect(nil, snapName+":network-bind", "")
+	utils.SnapConnect(nil, snapName+":custom-gpio", snapName+":custom-gpio-dev")
+	// chip-tool interfaces
+	utils.SnapConnect(nil, chipToolSnap+":avahi-observe", "")
```

Comment:
Some of these are only needed for the commissioning and control tests.

Reply:
Removed chip-tool installation and chip-tool interface connections from setup and used only in needed test. Thank you.